### PR TITLE
Prevent more than 8 players joining a game room

### LINKED ISSUE #54: Impose a constraint on the number of players in the room
We are only letting the game start if there are 2 to 8 players (which reminds me I should put that in the main README). It would only be logical that we don't allow a ninth person to join the room in the first place.

### PR DESCRIPTION
**Pull request by maciej-pomykala**
Closes #54

### CODE REVIEW

**Comment by adrian-golian** [comment]
Since you didn't respond to [this](https://github.com/Blef-team/blef_game_engine/pull/52#issuecomment-614340262) comment directly, could you let me know whether you can refactor the code to do the error-checks first and if none of them trigger then go on to the core function code?

**Comment by adrian-golian** [comment]
Separately, the api readme needs updating, if you're making changes to the API responses and codes.

**Comment by maciej-pomykala** [comment]
I made changes to `api/README.md` within the commit. D you mean the main `README.md`?

Regarding the refactorisation of the code, do you mean making the error codes and messages appear earlier in the `.R` file than the core function code?

**Comment by adrian-golian** [comment]
> I made changes to api/README.md within the commit. Do you mean the main README.md?
Oh, I don't know how I managed to read the changes but not register that they are the readme, apologies.

**Comment by adrian-golian** [comment]
> Regarding the refactorisation of the code, do you mean making the error codes and messages appear earlier in the .R file than the core function code?

I mean instead of this:
```
if (!checkA & !checkB & !checkC) {
  core function code
}
else if (checkA)  errorA
else if (checkB) errorB
else if (checkC) errorC
```
do
```
if (checkA)  errorA
if (checkB) errorB
if (checkC) errorC
core function code
```

**Comment by maciej-pomykala** [comment]
Sieht gut aus.

**Comment by adrian-golian** [comment]
That is if R, or the API framework, allows for such control flow - that the code will stop at `errorA` and exit the function.

**Comment by maciej-pomykala** [comment]
I can't believe it wouldn't, the code starts looking disgusting with all these `if`s

**Comment by maciej-pomykala** [comment]
The refactoring issue has been created. 
Anything else on this thing here?

File: api/endpoints.R [hunk 1]
```diff
@@ -48,22 +48,26 @@ function(game_uuid, nickname, res) {
...
         mutate(uuid = as.character(uuid), nickname = as.character(nickname))
       
       saveRDS(game, get_path(game_uuid))
       list(player_uuid = player_uuid)
+    } else if (game$status != "Not started") {
+      res$status <- 403
+      list(error = "Game already started")
+    } else if (game_full) {
+      res$status <- 403
+      list(error = "The game room is full")
```
Comment (adrian-golian) [inline api/endpoints.R:67]: Well, you could refactor the code added in this PR.

**Comment by maciej-pomykala** [comment]
Good idea - see latest commit for implementation.

**Comment by adrian-golian** [comment]
This is awesome. So much better.
One thing though - make sure you test it and so will I. I have no trust in R executing this code like we expect.

**Comment by maciej-pomykala** [comment]
Yeah, I tested that all errors appear, and that the ones to the top are prioritised

**Pull request merged**